test/e2e/config_test.go

```diff
@@ -71,6 +72,107 @@ func TestClusterMonitoringOperatorConfiguration(t *testing.T) {
 	f.AssertOperatorCondition(configv1.OperatorAvailable, configv1.ConditionTrue)(t)
 }
 
+func TestClusterMonitoringStatus(t *testing.T) {
+	const (
+		storage = "2Gi"
+	)
+	for _, tc := range []struct {
+		name               string
+		config             string
+		userWorkloadConfig string
+		assertion          func(t *testing.T)
+	}{
+		{
+			name:               "default config, no persistent storage",
+			config:             "",
+			userWorkloadConfig: "",
+			assertion: func(t *testing.T) {
+				f.AssertOperatorCondition(configv1.OperatorAvailable, configv1.ConditionTrue)(t)
+				f.AssertOperatorCondition(configv1.OperatorDegraded, configv1.ConditionFalse)(t)
+				f.AssertOperatorConditionReason(configv1.OperatorDegraded, client.StorageNotConfiguredReason)
+				f.AssertOperatorConditionMessage(configv1.OperatorDegraded, client.StorageNotConfiguredMessage)
+			},
+		},
+		{
+			name: "default config with presistent storage",
+			config: fmt.Sprintf(`prometheusK8s:
+  volumeClaimTemplate:
+    spec:
+      resources:
+        requests:
+        storage: %s
+`, storage),
+			userWorkloadConfig: "",
+			assertion: func(t *testing.T) {
+				f.AssertOperatorCondition(configv1.OperatorAvailable, configv1.ConditionTrue)(t)
+				f.AssertOperatorCondition(configv1.OperatorDegraded, configv1.ConditionFalse)(t)
+				f.AssertOperatorConditionReason(configv1.OperatorDegraded, "")
+				f.AssertOperatorConditionMessage(configv1.OperatorDegraded, "")
+			},
+		},
+		{
+			name: "default config with presistent storage but with UserAlermanagerConfig missconfiguration",
+			config: fmt.Sprintf(`alertmanagerMain:
```

Comment:
@simonpasquier I think I finally figured out the issue with this! It's here, I specified the CMO config but forgot to enable UWM 🙈 this somehow would make the test pass but it would put CMO tests in a weird state...